sdk/src/main/java/com/hedera/hashgraph/sdk/TransactionResponse.java

```diff
@@ -37,6 +44,15 @@
             .executeAsync(client);
     }
 
+    public TransactionRecord getRecordSync(Client client) throws Exception {
+        getReceiptSync(client);
```

Comment:
I don't understand why we need to get the receipt before getting a record. The record [contains](https://github.com/hashgraph/hedera-protobufs/blob/e0ee0797bacb0199fb1f2f28c42805ea0889f37c/services/transaction_record.proto#L43) the receipt. You should just be able to get the record by itself.

Reply:
@steven-sheehy This is the same as in the async method. The short reason is that we need to get the confirmed record. It costs nothing to poll a pending receipt until its confirmed. It would cost to poll a pending record.

Reply:
Ok, so it's purely a cost saving mechanism. Got it.